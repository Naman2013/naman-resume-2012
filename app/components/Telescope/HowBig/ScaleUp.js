import React, { Component } from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import domains from './domains';
import FadeSVG from '../../../components/common/Fade/FadeSVG';
import { animateValues } from '../../../utils/easingFunctions';

class ScaleUp extends Component {
  static BEFORE_START = 1000;

  static propTypes = {
    dimension: PropTypes.number,
    targetObjectURL: PropTypes.string.isRequired,
    targetObjectScale: PropTypes.number.isRequired,
    targetObjectName: PropTypes.string.isRequired,
    referenceObject: PropTypes.string.isRequired,
    onComplete: PropTypes.func,
  };

  static defaultProps = {
    dimension: 500,
    onComplete: noop,
  };

  state = {
    referenceObjectLoaded: false,
    showReference: false,
    referenceScale: 1,
    referencePosition: {
      x: 0,
      y: 0,
    },
  };

  componentDidMount() {
    this.beginDelayToShowReference();
  }

  componentWillUnmount() {
    clearTimeout(this.timerDelayPresentReference);
  }

  handleReferenceObjectLoaded = () => {
    this.setState({ referenceObjectLoaded: true });
  }

  beginDelayToShowReference() {
    setTimeout(() => {
      this.presentReference();
    }, ScaleUp.BEFORE_START);
  }

  presentReference() {
    this.setState({ showReference: true });
  }

  timerDelayPresentReference = undefined;

  render() {
    const { referenceObject } = this.props;
    const { showReference, referenceObjectLoaded } = this.state;

    const displayReferenceObject = (showReference && referenceObjectLoaded);

    return (
      <g>
        <FadeSVG isHidden={!(displayReferenceObject)}>
          {
            domains
              .enumValueOf(referenceObject)
              .render({
                width: 200,
                height: 200,
                x: 0,
                y: 0,
                onLoadCallback: this.handleReferenceObjectLoaded,
              })
          }
        </FadeSVG>
      </g>
    );
  }
}

export default ScaleUp;
