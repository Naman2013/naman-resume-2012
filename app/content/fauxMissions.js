export default {
  nonMission: { missionTargetID: 0 },
  scaleDown: {
    key: 'scaleDown',
    missionTargetID: 1,
    referenceObjectScale: 1,
    referenceObject: 'SOLAR_SYSTEM',
    targetObjectScale: 0.34,
    targetObjectURL: 'https://vega.slooh.com/icons/nav/Galaxy_nav_w.svg',
    targetObjectName: 'Test object',
  },
  scaleUp: {
    key: 'scaleUp',
    missionTargetID: 2,
    referenceObjectScale: 0.34,
    referenceObject: 'SOLAR_SYSTEM',
    targetObjectScale: 1,
    targetObjectURL: 'https://vega.slooh.com/icons/nav/menu-help.svg',
    targetObjectName: 'Test object',
  },
};
