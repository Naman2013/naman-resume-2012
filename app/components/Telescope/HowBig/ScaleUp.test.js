import React from 'react';
import { shallow } from 'enzyme';
import ScaleUp from './ScaleUp';
import fauxMissions from 'content/fauxMissions';

const setup = (customProps) => {
  const props = Object.assign({}, fauxMissions.scaleUp, customProps);
  const wrapper = shallow(<ScaleUp {...props} />);

  return {
    wrapper,
  };
};

describe('ScaleUp', () => {
  it('should render correctly', () => {
    expect(setup().wrapper).toMatchSnapshot();
  });
});
