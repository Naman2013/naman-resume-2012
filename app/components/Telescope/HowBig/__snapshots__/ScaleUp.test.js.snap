// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleUp rendering should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScaleUp
    dimension={500}
    missionTargetID={2}
    onComplete={[Function]}
    referenceObject="SOLAR_SYSTEM"
    referenceObjectScale={0.34}
    targetObjectName="Test object"
    targetObjectScale={1}
    targetObjectURL="https://vega.slooh.com/icons/nav/menu-help.svg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FadeSVG
        isHidden={true}
      >
        <ObjectFrame
          height={200}
          onLoadCallback={[Function]}
          svgURL={
            Object {
              "process": [Function],
            }
          }
          width={200}
          x={0}
          y={0}
        />
      </FadeSVG>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ObjectFrame
          height={200}
          onLoadCallback={[Function]}
          svgURL={
            Object {
              "process": [Function],
            }
          }
          width={200}
          x={0}
          y={0}
        />,
        "isHidden": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": 200,
          "onLoadCallback": [Function],
          "svgURL": Object {
            "process": [Function],
          },
          "width": 200,
          "x": 0,
          "y": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FadeSVG
          isHidden={true}
        >
          <ObjectFrame
            height={200}
            onLoadCallback={[Function]}
            svgURL={
              Object {
                "process": [Function],
              }
            }
            width={200}
            x={0}
            y={0}
          />
        </FadeSVG>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ObjectFrame
            height={200}
            onLoadCallback={[Function]}
            svgURL={
              Object {
                "process": [Function],
              }
            }
            width={200}
            x={0}
            y={0}
          />,
          "isHidden": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 200,
            "onLoadCallback": [Function],
            "svgURL": Object {
              "process": [Function],
            },
            "width": 200,
            "x": 0,
            "y": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
